### ISSUE TITLE
`smartcase` for filtering tabs

### ISSUE THREAD

**Issue by uu59** [ueokande/vim-vixen#125]
### I'm opening this issue because:

- [ ] I'll report a bug
- [x] I'll propose a new feature

### Description

Tab completion for `:buffer` is a case-sensitive now. But I'd like to filter them by case-insensitive keyword same as Vim's `:set smartcase`.

[tabs.selectByKeyword](https://github.com/ueokande/vim-vixen/blob/331b704d24158a25513a25540d2a5e82672606ec/src/background/tabs.js#L36) would be changed by that.

#### Steps to Reproduce

Please provide detailed steps for reproducing the issue.

1. Open https://issues.apache.org/jira/browse/YARN-2928
2. Type `:buffer`
3. Type `yarn`
4. Showing tab opened by step 1 in the completion list

**Comment by thiswillbeyourgithub** [comment]
I didn't see it proposed elsewhere : searching for text in the page with "/" is case sensitive too, it would be great to allow the user to choose if he wants it to be case insensitive, maybe via an argument in the settings file like : `"/": { "type": "find.start" , "case-sensitive": false},`

**Comment by qazip** [comment]
I agree, that an option in the settings would be good, but I think by default it should be case the buffer search should be case insensitive.

**Comment by kraml** [comment]
Yeah I think searching tab title or URL in the buffer command should by default be case insensitive, if the smart case feature won't be available soon. Searching and switching tab is not a serious task like text editing. It is for convenience and nobody actually distinguish tabs by upper or lower case of its title or URL... In fact I'm surprised when I found it it is not case insensitive.

**Comment by ueokande** [comment]
Currently, filtering tabs is case-sensitive.  But Pattern matching on the title is coming in Firefox 60.  Probably, it can close the issue.
https://bugzilla.mozilla.org/show_bug.cgi?id=1334782

Case-sensitive is disabled as default and fixed is better.  Case-sensitive for tabs is not so thankful for users, and I can keep code clean :D
